src/renderer/components/dock/log-tab.store.ts

```diff
@@ -167,18 +173,34 @@ export class LogTabStore extends DockTabStore<LogTabData> {
 
     if (error) {
       logger.warn(`[LOG-TAB-STORE]: data for ${tabId} was invalid`, error);
-      this.closeTab(tabId);
+      runInAction(() => {
+        this.closeTab(tabId);
+      });
```

Comment:
Actually that is incorrect, we do need the `runInAction` wrapper because otherwise mobx will try and update within the same event. And if that event is a render step then it throws an error.

However, I have refactored so that this code is no longer needed.